### PR TITLE
Azure Event Grid 2022-06-15 API version support

### PR DESCRIPTION
**Pull request by ahamad-MS**
In this PR, we are adding support for the new GA API version 2022-06-15. All the resources/operations included in this PR were already reviewed and approved by ARM team and were in use in previous preview API versions and not new ones. we simply promoting them to be ga. Here are list of preview API versions where these resources were added and used:
- https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2021-10-15-preview/EventGrid.json
- https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2021-06-01-preview/EventGrid.json
- https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-10-15-preview/EventGrid.json
- https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json

GA Features in this API version includes the following:

- Partner Events resources and operations, which includes:
1. partnernamespaces
2. partnernamespaces/channels
3. partnerRegistration
4. partnertopic
5. verifiedpartners
6. partnerconfiguration

- nested event subscriptions resources under, topics, domains, domain topics and partner topics.

- PrivateLinks/PrivateConnections for partner namespaces.

- dataResidencyBoundary for topic and domain

- Inline Eventtypes for channels and partner topics

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [X] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  8. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
06/09/2022

  9. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
06/13/2022

  10. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [X] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [X] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [X] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ahamad-MS** [comment]
@dw511214992 @ArcturusZhang Can you please help reviewing and approve these changes asap?

thanks

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2022-06-15/EventGrid.json [hunk 1]
```diff
@@ -0,0 +1,10924 @@
...
+            "description": "The type of the parent resource. This can be either \\'topics\\', \\'domains\\', or \\'partnerNamespaces\\'.",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "topics",
+              "domains",
+              "partnerNamespaces"
+            ],
+            "x-ms-enum": {
+              "name": "ParentType",
```
Comment (ArcturusZhang) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2022-06-15/EventGrid.json:6552]: Please align all the occurrence of this enum type name to "PrivateEndpointConnectionsParentType" with previous api-versions, such like https://github.com/Azure/azure-rest-api-specs/blob/2d6cb29af754f48a08f94cb6113bb1f01a4e0eb9/specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2021-12-01/EventGrid.json#L2642
This change of name is introducing an unneccesary breaking change to all the SDKs. And the old name is more descriptive

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

**Review by ArcturusZhang — APPROVED** [review]

**Comment by tadelesh** [comment]
@JeffreyRichter Could you help to review and approve this `x-ms-enum` change breaking, then we can continue with the SDK release? We've discussed it through mail and swagger tooling team will going to add rule for such change.

**Comment by JeffreyRichter** [comment]
Changing a property from a string to an enum-modeled-as-string does break SDKs because the method parameters and fields that used to be a string will now become a different data type preventing existing customer code from running/compiling.

Are you saying we want to make this breaking change now?
If so, then why?

**Comment by tadelesh** [comment]
This property is an enum before but without `x-ms-enum`. This version adds the extension and set `model-as-string: true`. It will cause the enum name change in Go SDK which is a breaking. I only know it is a fix for S360. @ahamad-MS Could you help to explain why we set the `model-as-stirng` flag?

**Comment by tadelesh** [comment]
Got it. It's a fix for S360 and this enum was handled as string before. @JeffreyRichter Could you help to add an approval for this breaking?

**Pull request merged**